### PR TITLE
feat(git): repo discovery + create worktree on disk

### LINKED ISSUE #7: vs-01c: Git repo discovery + create worktree on disk
## Parent PRD

#1

## What to build

Implement core git operations in `src/git/`:

1. **Repository discovery** — given a path, find the `.git` directory (walk up), extract repo name, remote URL, and default branch using `git2`. See FR-5.

2. **Create worktree** — given a branch name and base branch, use `git2` to create a new git worktree at the path resolved by the template renderer (from #3). Creates the branch from base if it doesn't exist locally. See FR-5, FR-6.

All git operations use `git2` (libgit2 bindings) — no shelling out to `git`.

### Layers touched
git, paths, tests

## Acceptance criteria

- [ ] `discover_repo(path)` finds `.git` dir by walking up from `path`
- [ ] Returns typed `RepoInfo` struct: name, path, remote_url, default_branch
- [ ] `create_worktree(repo, branch, base, target_path)` creates a real git worktree via `git2`
- [ ] Branch is created from base if it doesn't exist locally
- [ ] Worktree directory exists on disk after creation
- [ ] `git worktree list` (via shell) shows the new worktree
- [ ] Integration tests use temp git repos (not the real repo)
- [ ] Error handling: returns typed errors for "not a git repo", "branch already exists"

## Blocked by

- #3

## User stories addressed

- US-1 (core worktree creation)

### PR DESCRIPTION
**Pull request by sadiksaifi**
Closes #7

## Summary
Implements git repository discovery (`discover_repo`) and worktree creation (`create_worktree`) in `src/git/mod.rs` using `git2` (libgit2 bindings). Discovery walks up from any path to find the `.git` directory and extracts repo name, path, origin remote URL, and default branch. Worktree creation validates branch uniqueness, creates a new branch from the specified base, and adds a git worktree at the target path on disk.

## User Stories Addressed
- US-1: Create a new worktree for a feature branch with one command

## Automated Testing
- Total tests added: 9 (in `src/git/mod.rs`)
- Tests passing: 9
- Tests failing: 0
- `cargo clippy`: pass (only pre-existing dead_code warnings from unused modules)
- `cargo test`: pass (67 total, 0 failures)

## UAT Checklist
- [ ] Clone the repo, run `cargo test git::tests` — all 9 git tests pass
- [ ] In a git repo, call `discover_repo(".")` from a Rust test/script — returns correct repo name, path, remote URL, and default branch
- [ ] Call `discover_repo` from a nested subdirectory — still finds the repo
- [ ] Call `discover_repo` on a plain (non-git) directory — returns `NotAGitRepo` error
- [ ] Call `create_worktree` with a new branch name — worktree directory exists on disk, `git worktree list` shows it
- [ ] Call `create_worktree` with an already-existing branch — returns `BranchAlreadyExists` error, no directory created
- [ ] Verify created worktree has a `.git` file (not directory) and the branch points to the same commit as base

## Known Issues
None

## Summary by CodeRabbit

* **New Features**
  * Repository discovery to surface repo metadata (name, path, remote, default branch)
  * Create worktrees with branch creation from an existing base and configurable target location

* **Bug Fixes**
  * Ensures branches/worktrees are cleaned up if worktree creation fails to avoid leftover state

### CODE REVIEW

File: src/git/mod.rs [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+        })?
+        .canonicalize()
+        .map_err(|_| GitError::NotAGitRepo {
+            path: path.to_path_buf(),
+        })?;
+
+    let name = workdir
+        .file_name()
+        .map(|n| n.to_string_lossy().into_owned())
+        .unwrap_or_default();
```
[the remark this answers is not in the document]
Comment (sadiksaifi) [reply src/git/mod.rs]: @coderabbitai Fixed in 746a19d.

- Changed `.unwrap_or_default()` → `.unwrap_or_else(|| String::from("repo"))` so name is never empty.
- Added `discover_repo_name_is_nonempty` regression test.

File: src/git/mod.rs [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+
+    // Resolve base branch to a commit
+    let base_ref = repo.find_branch(base, git2::BranchType::Local)?;
+    let base_commit = base_ref.get().peel_to_commit()?;
+
+    // Create the new branch from base and add the worktree
+    let new_branch = repo.branch(branch, &base_commit, false)?;
+    let mut opts = git2::WorktreeAddOptions::new();
+    opts.reference(Some(new_branch.get()));
+    repo.worktree(branch, target_path, Some(&opts))?;
```
[the remark this answers is not in the document]
Comment (sadiksaifi) [reply src/git/mod.rs]: @coderabbitai Fixed in 746a19d.

- Worktree creation is now wrapped in a block scope; on failure, the orphaned branch is deleted before re-propagating the error.
- Added `create_worktree_cleans_up_branch_on_worktree_failure` test that places a blocker file at `target_path`, asserts the error, and verifies the branch is cleaned up.

**Comment by sadiksaifi** [comment]
@coderabbitai All 6 review findings addressed in 746a19d:

| # | Finding | Fix |
|---|---------|-----|
| 1 | `PartialEq` on `RepoInfo` (nitpick, L3-10) | Added `PartialEq` derive + `repo_info_supports_equality` test |
| 2 | Empty name fallback (minor, L44-47) | `.unwrap_or_default()` → `.unwrap_or_else(\|\| String::from("repo"))` + regression test |
| 3 | Error context lost in `discover` (nitpick, L29-32) | Extracted `map_repo_open_error` helper — `NotFound` → `NotAGitRepo`, else → `Git(e)` |
| 4 | Error context lost in `open` (nitpick, L82-84) | Same helper applied to `Repository::open` |
| 5 | Orphaned branch on worktree failure (major, L100-104) | Branch deleted on worktree error + cleanup test |
| 6 | Missing edge case tests (nitpick, L131-287) | Added `create_worktree_errors_when_base_branch_does_not_exist` + `create_worktree_errors_when_target_path_already_exists` |

All 16 tests pass (9 existing + 7 new), no new clippy warnings.
